### PR TITLE
Awesome books: refactor  use  classes

### PR DESCRIPTION
**Pull request by mohamedgamaleldin999999**
in this pr, We are trying to refactor the previous code so we use classes instead

### CODE REVIEW

**Review by V-Blaze — CHANGES_REQUESTED** [review]
Hi @Jessekay, @ClarenceAtim @mohamedgamaleldin999999 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

dcentl8-9b96e129-9725-401b-ac34-2feb990202a0

## Invalid Code Review Request

## You have submitted a project with incomplete Microverse important headers, According to Microverse all important sections in the readme file must be present, Your project readme is missing the MIT license section, I'll suggest you add this section to your readme and add the appropriate content too. :+1: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: index.js [hunk 1]
```diff
@@ -1,58 +1,64 @@
...
-// Render the book list on page load
-renderBookList();
-
-document.querySelector('form').addEventListener('submit', () => {
-  this.reset();
-});
\ No newline at end of file
+  updateLocalStorage() {
+    localStorage.setItem('books', JSON.stringify(this.books));
+  }
```
Comment (abdulhamiid) [inline index.js:62]: According to the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m2_plain_js_classes_v1_1.md), it is important that we maintain all of the original functionality of the application (i.e. no bugs were introduced while refactoring). 👀

Kindly ensure that data is preserved in the browser's memory by using [localStorage](https://github.com/microverseinc/curriculum-html-css/blob/main/local_storage.md). 🤗👍

File: index.html [hunk 2]
```diff
@@ -9,11 +9,12 @@
...
-    <input type="text" id="title" name="title" placeholder="Title" required><br><br>
-    <input type="text" id="author" name="author" placeholder="Author" required><br><br>
-    <button type="submit">Add</button>
-  </form>
+
+    <form id="add-book-form">
+      <input type="text" id="title-input" placeholder="Title" required>
+      <input type="text" id="author-input" placeholder="Author" required>
+      <button type="submit">Add Book</button>
+    </form>
```
Comment (abdulhamiid) [inline index.html]: According to the [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m2_plain_js_classes_v1_1.md), kindly use CSS styles to make your application match [this wireframe](https://github.com/microverseinc/curriculum-javascript/blob/bb245ca8a9d3bb04ebefbbd45e800764e1fc5f74/books/images/awesome_books_core_elements.png) 👍

Your design
image

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Linters checks are passing
- ✔️ Use JavaScript classes.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 2 above)
Comment (Qoosim) [inline index.html]: - [x] The required change has not been implemented. Kindly attend to it and make your design looks like the wireframe as required from the  [project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/m2_plain_js_classes_v1_1.md).

### Your Design
your-design

### Wireframe
wireframe

File: index.js (hunk 1 above)
Comment (Qoosim) [inline index.js:62]: - [x] Kindly look into the way you setItem and getItem to and from your localStorage. Your localStorage reflects newly created books and deleted  books, but the books get disappeared on the browser the the page is refreshed. Kindly fix this.

### App Behaviour
[your-app.webm](https://github.com/Jessekay/AwesomeBooks/assets/26597432/93f833f5-903b-4a52-b7a5-8bf2884df136)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [x] Kindly ensure your PR Title reflect the milestone you're working on. Your current PR title does not tally with the milestone you worked on. Please, always keep this in mind.

pr-title


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 2 above)
Comment (ClarenceAtim) [reply index.html]: Hi sir thanks for your feedback
DO  mean we should produce the same design or we can use our design and make it similar to this?

**Review by ghost — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ App works as expected

## Optional suggestions

N/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**